src/algorithms/synthesis/syrec_synthesis.cpp

```diff
@@ -0,0 +1,1447 @@
+            if (!full_statement(stat)) {
+                if (!on_statement(stat)) {
+                    return false;
+                }
+            }
+        }
+
+        modules.pop();
+
+        return true;
+    }
+
+    bool standard_syrec_synthesizer::on_statement(const applications::uncall_statement& statement) {
+        using boost::adaptors::reversed;
+        using boost::adaptors::transformed;
+
+        // 1. Adjust the references module's parameters to the call arguments
+        for (unsigned i = 0u; i < statement.parameters().size(); ++i) {
+            const std::string& parameter        = statement.parameters().at(i);
+            const auto&        module_parameter = statement.target()->parameters().at(i);
+
+            module_parameter->set_reference(modules.top()->find_parameter_or_variable(parameter));
+        }
+
+        // 2. Create new lines for the module's variables
+        add_variables(_circ, statement.target()->variables());
+
+        modules.push(statement.target());
+        for (const auto& stat: statement.target()->statements() | reversed | transformed(syrec::applications::reverse_statements())) {
+            if (!full_statement(stat)) {
+                if (!on_statement(stat)) {
+                    return false;
+                }
+            }
+        }
+
+        modules.pop();
+
+        return true;
+    }
+
+    bool standard_syrec_synthesizer::on_statement(const applications::skip_statement& statement [[maybe_unused]]) {
+        return true;
+    }
+
+    bool standard_syrec_synthesizer::on_expression(const applications::expression::ptr& expression, std::vector<unsigned>& lines, std::vector<unsigned>& lhs_stat, unsigned op) {
+        if (auto* exp = dynamic_cast<applications::numeric_expression*>(expression.get())) {
+            return on_expression(*exp, lines, lhs_stat, op);
+        } else if (auto* exp_3 = dynamic_cast<applications::variable_expression*>(expression.get())) {
+            return on_expression(*exp_3, lines, lhs_stat, op);
+        } else if (auto* exp_4 = dynamic_cast<applications::binary_expression*>(expression.get())) {
+            return on_expression(*exp_4, lines, lhs_stat, op);
+        } else if (auto* exp_5 = dynamic_cast<applications::shift_expression*>(expression.get())) {
+            return on_expression(*exp_5, lines, lhs_stat, op);
+        } else {
+            return false;
+        }
+    }
+
+    bool standard_syrec_synthesizer::on_expression(const applications::numeric_expression& expression, std::vector<unsigned>& lines, std::vector<unsigned>& lhs_stat [[maybe_unused]], unsigned op [[maybe_unused]]) {
+        return get_constant_lines(expression.bitwidth(), expression.value()->evaluate(loop_map), lines);
+    }
+
+    bool standard_syrec_synthesizer::on_expression(const applications::variable_expression& expression, std::vector<unsigned>& lines, std::vector<unsigned>& lhs_stat [[maybe_unused]], unsigned op [[maybe_unused]]) {
```

Comment:
if the parameters `las_stat` and `op` are not used by the function, they should not be part of its signature.